llama-index-core/llama_index/core/query_pipeline/query.py

```diff
@@ -365,8 +393,36 @@ def run_multi(
                 CBEventType.QUERY,
                 payload={EventPayload.QUERY_STR: json.dumps(module_input_dict)},
             ) as query_event:
-                outputs, _ = self._run_multi(module_input_dict)
-                return outputs
+                if batch:
+                    outputs = {}
+
+                    batch_size = min(
```

Comment:
Ha you weren't kidding about this being complicated.

Should be validating that all the inputs have the same batch size?